modules/cfe_testcase/src/es_info_test.c

```diff
@@ -0,0 +1,261 @@
+    UtAssert_True(strlen(ESAppInfo.EntryPoint) == 0, "ES App Info -> EntryPt  = %s", ESAppInfo.EntryPoint);
+
+    UtAssert_StrCmp(TestAppInfo.FileName, "/cf/cfe_testrunner.so", "Test App Info -> FileName = %s",
+                    TestAppInfo.FileName);
+    UtAssert_True(strlen(ESAppInfo.FileName) == 0, "ES App Info -> FileName  = %s", ESAppInfo.FileName);
+
+    UtAssert_True(TestAppInfo.StackSize > 0, "Test App Info -> StackSz  = %d", TestAppInfo.StackSize);
+    UtAssert_True(ESAppInfo.StackSize > 0, "ES App Info -> StackSz  = %d", ESAppInfo.StackSize);
+
+    if (TestAppInfo.AddressesAreValid)
+    {
+        UtAssert_True(TestAppInfo.AddressesAreValid > 0, "Test App Info -> AddrsValid?  = %d",
+                      TestAppInfo.AddressesAreValid);
+        UtAssert_True(TestAppInfo.CodeAddress > 0, "Test App Info -> CodeAddress  = %d", TestAppInfo.CodeAddress);
+        UtAssert_True(TestAppInfo.CodeSize > 0, "Test App Info -> CodeSize  = %d", TestAppInfo.CodeSize);
+        UtAssert_True(TestAppInfo.DataAddress > 0, "Test App Info -> DataAddress  = %d", TestAppInfo.DataAddress);
+        UtAssert_True(TestAppInfo.DataSize > 0, "Test App Info -> DataSize  = %d", TestAppInfo.DataSize);
+        UtAssert_True(TestAppInfo.BSSAddress > 0, "Test App Info -> BSSAddress  = %d", TestAppInfo.BSSAddress);
+        UtAssert_True(TestAppInfo.BSSSize > 0, "Test App Info -> BSSSize  = %d", TestAppInfo.BSSSize);
+    }
+    else
+    {
+        UtAssert_NA("AddressesAreValid is false: Code Address test skipped");
+        UtAssert_NA("AddressesAreValid is false: Code Size test skipped");
+        UtAssert_NA("AddressesAreValid is false: Data Address test skipped");
+        UtAssert_NA("AddressesAreValid is false: Data Size test skipped");
+        UtAssert_NA("AddressesAreValid is false: BSS  Address test skipped");
+        UtAssert_NA("AddressesAreValid is false: BSS Size test skipped");
+    }
+
+    UtAssert_True(ESAppInfo.AddressesAreValid == 0, "ES App Info -> AddrsValid?  = %d", ESAppInfo.AddressesAreValid);
+
+    UtAssert_True(TestAppInfo.StartAddress > 0, "Test App Info -> StartAddress  = 0x%4x", TestAppInfo.StartAddress);
+    UtAssert_True(ESAppInfo.StartAddress == 0, "ES App Info -> StartAddress  = 0x%4x", ESAppInfo.StartAddress);
+
+    UtAssert_INT32_EQ(TestAppInfo.ExceptionAction, 0);
+    UtAssert_INT32_EQ(ESAppInfo.ExceptionAction, 1);
+
+    UtAssert_True(TestAppInfo.Priority > 0, "Test App Info -> Priority  = %d", TestAppInfo.Priority);
+    UtAssert_True(ESAppInfo.Priority > 0, "ES App Info -> Priority  = %d", ESAppInfo.Priority);
+
+    UtAssert_True(TestAppInfo.NumOfChildTasks == 0, "Test App Info -> Child Tasks  = %d", TestAppInfo.NumOfChildTasks);
+    UtAssert_True(ESAppInfo.NumOfChildTasks > 0, "ES App Info -> Child Tasks  = %d", ESAppInfo.NumOfChildTasks);
+
+    UtAssert_INT32_EQ(CFE_ES_GetAppIDByName(&AppIdByName, InvalidName), CFE_ES_ERR_NAME_NOT_FOUND);
+    UtAssert_INT32_EQ(AppIdByName, CFE_ES_APPID_UNDEFINED);
```

Comment:
Should not test as [u]int32 - use `!CFE_RESOURCEID_TEST_DEFINED()` macro

Reply:
@jphickey same as above, do we really need to change 
UtAssert_INT32_EQ(AppIdByName, CFE_ES_APPID_UNDEFINED);
to
UtAssert_True(!CFE_RESOURCEID_TEST_DEFINED(AppIdByName), "AppIdByName (%d) = CFE_ES_APPID_UNDEFINED (%d)", AppIdByName, CFE_ES_APPID_UNDEFINED);

Reply:
Note that you'll need to use `CFE_RESOURCEID_TO_ULONG` macro and the `%lu` conversion specifier as I mentioned in my previous comment.  Please enable the `set(MISSION_RESOURCEID_MODE STRICT)` in your build as I mentioned, so the compiler will better tell you if you've made type assumptions or mixed together types like this.

We also shouldn't assume that `CFE_RESOURCEID_TEST_DEFINED` is the same as equality with the `CFE_ES_APPID_UNDEFINED` value.  There could be many values which are "not defined", of which `CFE_ES_APPID_UNDEFINED` is guaranteed to be one of them, but maybe not the other way around.   It's like a square/rectangle type of thing.

Reply:
It was your idea to use CFE_RESOURCEID_TEST_DEFINED. Since I am comparing ID's I rather use CFE_RESOURCEID_TEST_EQUAL if we can't just UtAssert_INT32_EQ

Reply:
No, this test should not be checking for equality to a specific value.  The only thing that is ensured is that if `CFE_ES_GetAppIDByName()` fails and returns an error, that the output ID also will also __not__ pass the `CFE_RESOURCEID_TEST_DEFINED` check.

Yes, in the current implementation this outputs specifically the `CFE_ES_APPID_UNDEFINED` value, but that might not always be the case, and might not be the only possible value.  

So the only thing that should be checked/asserted here is that `CFE_RESOURCEID_TEST_DEFINED()` is false.  The actual value doesn't matter - we don't need to compare it to a specific value.  Just confirm that the test is false, and that's enough.

Reply:
To clarify I would recommend the assert as follows:

    UtAssert_True(!CFE_RESOURCEID_TEST_DEFINED(AppIdByName), "AppIdByName (%lu) not defined", CFE_RESOURCEID_TO_ULONG(AppIdByName));

Also note that I have no objections to making a local macro or inline function to do this check as this is likely to need to be repeated frequently, and it would make it more readable.

Reply:
I disagree, since the code now sets it to CFE_ES_APPID_UNDEFINED that is what it should be testing for. If right now it was set to something different that was also invalid, I would view that as a bug since the code is not doing what you expect it to and so should be treated as a bug. If in the future someone changes the functionality of CFE_ES_GetAppIDByName then the test would be changed to test the new functionality.

Reply:
Are we good with this?